javascript-algorithms-and-data-structures/learn-basic-string-and-array-methods-by-building-a-music-player-app/scripts.js

```diff
@@ -0,0 +1,231 @@
+
+const playSVG = document.getElementById("play");
+
+function playSong(song) {
+    const oldIndex = currentIndex;
+    // using indesOf method
+    const newIndex = songs.indexOf(song);
+
+    playSVG.style.fill="#f1be32";
+
+    if (oldIndex !== newIndex) {
+        clearSongBgs();
+        updatePlayingSongBackground(newIndex)
+    }
+
+    audio.src = song.src;
+    audio.title = song.title;
+    audio.currentTime = currentTime;
+    document.getElementById("player__song-title").innerText = song.title;
+    document.getElementById("player__song-artist").innerText = song.artist;
+    document.querySelector(".play").ariaCurrent = audio.title;
+    document.getElementById("player__album-art").innerHTML = `<img src="${song.cover}" alt="song cover art" />`;
+    audio.play();
+    currentIndex = newIndex;
+
+    updatePlayingSongBackground(newIndex);
+
+}
+
+function pauseSong() {
+  audio.pause();
+  currentTime = audio.currentTime;
+  playSVG.style.fill="#fbf9f9";
+}
+
+function togglePlay() {
+  playSong(songs[currentIndex]);
+  updatePlayingSongBackground(currentIndex)
+
+}
+
+function toggleClosePlaylist() {
+    if (songListDiv.style.visibility == "hidden") {
+        songListDiv.style.visibility = "visible";
+        songListDiv.style.display = "flex";
+        closeBtn.setAttribute('aria-expanded', true);
+
+    } else {
+        songListDiv.style.visibility = "hidden";
+        songListDiv.style.display = "none";
+        closeBtn.setAttribute('aria-expanded', false);
+    }
+}
+
+// player controls
+function nextSong() {
+  const nextIndex = (currentIndex + 1) % songs.length;
+  playSong(songs[nextIndex]);
+  currentIndex = nextIndex;
+
+//   keeping keyboard indicator in sync with array keyls/selected indicator.
+  if(document.activeElement.classList.contains('playlist__song')) {
+    const songDivs = document.querySelectorAll(".playlist__song");
+    songDivs[currentIndex].focus();
+  }
+}
+
+function previousSong() {
+  const prevIndex = (currentIndex - 1 + songs.length) % songs.length;
+  playSong(songs[prevIndex]);
+  currentIndex = prevIndex;
+
+//   keeping keyboard indicator in sync with array keyls/selected indicator.
+  if(document.activeElement.classList.contains('playlist__song')) {
+      const songDivs = document.querySelectorAll(".playlist__song");
+      songDivs[currentIndex].focus();
+    }
+}
+
+// playlist display
+function clearSongBgs() {
+    document.querySelectorAll(".playlist__song").forEach(song => {
+        song.removeAttribute('aria-current');
+    });
+}
+
+function updatePlayingSongBackground(newIndex) {
+    const songDivs = document.querySelectorAll(".playlist__song");
+    songDivs[newIndex].setAttribute('aria-current', true);
+}
+
+function playSelectedSong(song, songBtn) {
+    playSong(song);
+    clearSongBgs();
+    songBtn.setAttribute('aria-current', true);
+}
+
+function shuffle() {
+    // Shallow copy done with spread operator
+    let songsCopy = [...songs];
```

Comment:
I believe this is the first time the spread operator is being introduced. 
So it would be helpful to make a note of that here when we break it down into steps and tests

```suggestion
    // first time the spread operator is being taught. Make sure to introduce with example in the steps
    let songsCopy = [...songs];
```